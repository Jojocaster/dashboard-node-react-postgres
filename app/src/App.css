a {
  color: inherit;
  text-decoration: none;
}

.app {
  display: flex;
  height: 100%;
  text-align: center;
}

.app-content {
  background-color: #FAFAFA;
  flex-grow: 1;
}

.app-logo {
  animation: app-logo-spin infinite 20s linear;
  height: 80px;
}

.app-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}

.app-title {
  font-size: 1.5em;
}

.app-intro {
  font-size: large;
}

@keyframes app-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}
