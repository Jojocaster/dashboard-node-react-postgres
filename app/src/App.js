import React, { Component } from 'react';
import {BrowserRouter as Router,Route} from 'react-router-dom';

import routes from './routes/';

import logo from './logo.svg';
import './app.css';

import Sidebar from './components/Sidebar/';

class App extends Component {
  render() {
    return (
      <Router>
        <div className="app">
          <Sidebar/>
          <section className="app-content">
            {routes.map((route, i) => (
              <Route key={i} exact={route.exact} path={route.path} component={route.component} />
            ))}
          </section>
        </div>
      </Router>
    );
  }
}

export default App;