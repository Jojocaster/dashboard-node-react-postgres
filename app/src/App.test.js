import React from 'react';
import {shallow, mount} from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import App from './app';

describe('App', () => {
  // const wrapper = mount(<App />);
  
  it('renders without crashing', () => {
    const wrapper = shallow(<App />); 
    expect(wrapper).toMatchSnapshot();
  });

  it('should have a main container', () => {
    const wrapper = shallow(<App />);
    expect(wrapper.find('.app').length).toBe(1);
  })

  it('should have one Sidebar component', () => {
    const wrapper = mount(<App />);
    expect(wrapper.find('.sidebar').length).toBe(1);
  })

  it('should have on section', () => {
    const wrapper = shallow(<App />);
    expect(wrapper.find('section').length).toBe(1);
  })
})
  