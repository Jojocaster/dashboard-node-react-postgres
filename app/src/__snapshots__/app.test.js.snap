// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div
        className="app"
      >
        <Sidebar />
        <section
          className="app-content"
        >
          <Route
            component={[Function]}
            path="/todos/:id?"
          />
        </section>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Sidebar />,
          <section
            className="app-content"
          >
            <Route
              component={[Function]}
              path="/todos/:id?"
            />
          </section>,
        ],
        "className": "app",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Route
                component={[Function]}
                path="/todos/:id?"
              />,
            ],
            "className": "app-content",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/todos/:id?",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "section",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="app"
        >
          <Sidebar />
          <section
            className="app-content"
          >
            <Route
              component={[Function]}
              path="/todos/:id?"
            />
          </section>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Sidebar />,
            <section
              className="app-content"
            >
              <Route
                component={[Function]}
                path="/todos/:id?"
              />
            </section>,
          ],
          "className": "app",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Route
                  component={[Function]}
                  path="/todos/:id?"
                />,
              ],
              "className": "app-content",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "0",
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "path": "/todos/:id?",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "section",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
